sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/src/LruCache.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Data.SchemaRegistry.ApacheAvro
+{
+    internal class LruCache<K,V>
+    {
+        private readonly int _capacity;
+        private readonly LinkedList<KeyValuePair<K, V>> _linkedList;
+        private readonly Dictionary<K, LinkedListNode<KeyValuePair<K, V>>> _map;
+        private readonly object _syncLock;
+
+        public LruCache(int capacity)
+        {
+            _capacity = capacity;
+            _linkedList = new LinkedList<KeyValuePair<K, V>>();
+            _map = new Dictionary<K, LinkedListNode<KeyValuePair<K, V>>>();
+            _syncLock = new object();
+        }
+
+        public bool TryGet(K key, out V value)
+        {
+            lock (_syncLock)
+            {
+                if (_map.TryGetValue(key, out var node))
+                {
+                    value = node.Value.Value;
+                    _linkedList.Remove(node);
+                    _linkedList.AddLast(node);
+                    return true;
+                }
+
+                value = default(V);
+                return false;
+            }
+        }
+
+        public void AddOrUpdate(K key, V val)
+        {
+            lock (_syncLock)
+            {
+                if (_map.TryGetValue(key, out var existingNode))
```

Comment:
This path doesn't make use of `val`,  is that intentional?   Other than moving the ordering around for the key, there's no update taking place?

Reply:
Ah good catch - not intentional. We should update it if not equal.